src/librustc_target/abi/call/mod.rs

```diff
@@ -380,6 +382,25 @@ impl<'a, Ty> ArgType<'a, Ty> {
         }
     }
 
+    pub fn make_unsized_indirect(&mut self, vtable_size: Option<Size>) {
+        self.make_indirect();
```

Comment:
You can move this whole function in `make_indirect` and have it look at `self.layout` to determine whether the type is unsized and whatnot.